modules/ds18b20/ds18b20.c

```diff
@@ -0,0 +1,433 @@
+    if (!OneWireStruct)   return DS18B20_USAGE_ERROR;
+    if (!ROM)             return DS18B20_USAGE_ERROR;
+    if (!DS18B20_Is(ROM)) return DEVICE_NOT_DS18B20;
+
+    char trigger_register_lo, trigger_register_hi, conf_register;
+    if (temp > DS18B20_MAX_TEMP_DEG_C) {
+        temp = DS18B20_MAX_TEMP_DEG_C;
+    } 
+    if (temp < DS18B20_MIN_TEMP_DEG_C) {
+        temp = DS18B20_MIN_TEMP_DEG_C;
+    }
+    /* Reset line */
+    OneWire_Reset(OneWireStruct);
+    /* Select ROM number */
+    OneWire_SelectWithPointer(OneWireStruct, ROM);
+    /* Read scratchpad command by OneWireStruct protocol */
+    OneWire_WriteByte(OneWireStruct, ONEWIRE_CMD_RSCRATCHPAD);
+    
+    /* Ignore first 2 bytes */
+    OneWire_ReadByte(OneWireStruct);
+    OneWire_ReadByte(OneWireStruct);
+    
+    trigger_register_hi = OneWire_ReadByte(OneWireStruct);
+    trigger_register_lo = OneWire_ReadByte(OneWireStruct);
+    conf_register = OneWire_ReadByte(OneWireStruct);
+    
+    trigger_register_lo = temp; 
+
+    /* Reset line */
+    OneWire_Reset(OneWireStruct);
+    /* Select ROM number */
+    OneWire_SelectWithPointer(OneWireStruct, ROM);
+    /* Write scratchpad command by OneWireStruct protocol, only trigger_register_hi, trigger_register_lo and conf_register register can be written */
+    OneWire_WriteByte(OneWireStruct, ONEWIRE_CMD_WSCRATCHPAD);
+    
+    /* Write bytes */
+    OneWire_WriteByte(OneWireStruct, trigger_register_hi);
+    OneWire_WriteByte(OneWireStruct, trigger_register_lo);
+    OneWire_WriteByte(OneWireStruct, conf_register);
+    
+    /* Reset line */
+    OneWire_Reset(OneWireStruct);
+    /* Select ROM number */
+    OneWire_SelectWithPointer(OneWireStruct, ROM);
+    /* Copy scratchpad to EEPROM of DS18B20 */
+    OneWire_WriteByte(OneWireStruct, ONEWIRE_CMD_CPYSCRATCHPAD);
+    
+    return 1;
+}
+
+char DS18B20_SetAlarmHighTemperature(OneWire_t* OneWireStruct, uint8_t *ROM, int8_t temp) {
+    if (!OneWireStruct)   return DS18B20_USAGE_ERROR;
+    if (!ROM)             return DS18B20_USAGE_ERROR;
+    if (!DS18B20_Is(ROM)) return DEVICE_NOT_DS18B20;
+
+    char trigger_register_lo, trigger_register_hi, conf_register;
+
+    if (temp > DS18B20_MAX_TEMP_DEG_C) {
+        temp = DS18B20_MAX_TEMP_DEG_C;
+    } 
+    if (temp < DS18B20_MIN_TEMP_DEG_C) {
+        temp = DS18B20_MIN_TEMP_DEG_C;
+    }
+    /* Reset line */
+    OneWire_Reset(OneWireStruct);
+    /* Select ROM number */
+    OneWire_SelectWithPointer(OneWireStruct, ROM);
+    /* Read scratchpad command by OneWireStruct protocol */
+    OneWire_WriteByte(OneWireStruct, ONEWIRE_CMD_RSCRATCHPAD);
+    
+    /* Ignore first 2 bytes */
+    OneWire_ReadByte(OneWireStruct);
+    OneWire_ReadByte(OneWireStruct);
+    
+    trigger_register_hi = OneWire_ReadByte(OneWireStruct);
+    trigger_register_lo = OneWire_ReadByte(OneWireStruct);
+    conf_register = OneWire_ReadByte(OneWireStruct);
+    
+    trigger_register_hi = temp; 

```

Comment:
trigger_register_hi gets assigned from a call to OneWire_ReadByte() but then gets assigned to temp without being used.  The implication here is that line 347 doesn't need to assign trigger_register_hi.  Is that indeed the case?  If so, why not just do a call to OneWire_ReadByte() without assignment to ignore the byte (like lines 343 - 345)?